projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx

```diff
@@ -80,15 +72,21 @@ export class DotFormComponent {
         return (
             <form onSubmit={(evt: Event) => this.handleSubmit(evt)}>
                 {this.fields.map((field: DotCMSContentTypeField) => this.getField(field))}
-                <button type='submit' disabled={this._valid ? null : true }>{this.submitLabel}</button>
-                <button type='button' onClick={() => this.resetForm()} >{this.resetLabel}</button>
+                <button type='submit' disabled={this.status.dotValid ? null : true}>
+                    {this.submitLabel}
+                </button>
+                <button type='button' onClick={() => this.resetForm()}>
+                    {this.resetLabel}
+                </button>
             </form>
         );
     }
 
     private updateFieldsValues() {
         this.fields = this.fields.map((field: DotCMSContentTypeField) => {
-            return typeof this.value[field.variable] !== 'undefined' ? { ...field, defaultValue: this.value[field.variable] } : field;
+            return typeof this.value[field.variable] !== 'undefined'
```

Comment:
Are we checking `undefined` because it could be `false`?

Reply:
this function is called to update the fields value obj that comes from the BE in the form. 

so there could be a `field` that comes from the BE that does not contains the `variable` param, in that case we return the `field` as is.

on the other hand if it has the param `variable` then we update the `fields` array with the latest change

Reply:
after debugging `typeof this.value[field.variable] !== 'undefined'` is still needed, when you do `reset` and the `field` contains a `defaultValue` you should clean up the field and not set the `defaultValue`.

So at the moment that condition is evaluated the value is `''` it will set the `defaultValue` which is wrong